Client/Application/WebServerUtil.swift

```diff
@@ -0,0 +1,90 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import GCDWebServers
+import Shared
+
+class WebServerUtil {
+
+    private var readerModeHander: ReaderModeHandlersProtocol
+    private var webServer: WebServerProtocol
+    private var profile: Profile
+
+    init(readerModeHander: ReaderModeHandlersProtocol = ReaderModeHandlers(),
+         webServer: WebServer = WebServer.sharedInstance,
+         profile: Profile) {
+        self.readerModeHander = readerModeHander
+        self.webServer = webServer
+        self.profile = profile
+    }
+
+    func setUpWebServer() {
+        guard !webServer.server.isRunning else { return }
+
+        readerModeHander.register(webServer, profile: profile)
+
+        let responders: [(String, InternalSchemeResponse)] =
+             [(AboutHomeHandler.path, AboutHomeHandler()),
+              (AboutLicenseHandler.path, AboutLicenseHandler()),
+              (SessionRestoreHandler.path, SessionRestoreHandler()),
+              (ErrorPageHandler.path, ErrorPageHandler())]
+        responders.forEach { (path, responder) in
+            InternalSchemeHandler.responders[path] = responder
+        }
+
+        if AppConstants.IsRunningTest || AppConstants.IsRunningPerfTest {
+            registerHandlersForTestMethods(server: webServer.server)
+        }
+
+        // Bug 1223009 was an issue whereby CGDWebserver crashed when moving to a background task
+        // catching and handling the error seemed to fix things, but we're not sure why.
+        // Either way, not implicitly unwrapping a try is not a great way of doing things
+        // so this is better anyway.
+        do {
+            try webServer.start()
```

Comment:
Just a thought - not a PR blocker at all.

I know you didn't author it, but is this something we should be more concerned about? `start` is discardable and wrapped in a try catch without documentation explaining why a log statement is enough to continue. I guess we can investigate the consequence further in a future ticket (and maybe if we can defer this to a later point in app init!)